akka-http2-support/src/main/scala/akka/http/impl/engine/http2/WrappedSslContextSPI.scala

```diff
@@ -6,49 +6,87 @@ package akka.http.impl.engine.http2
 
 import java.security.SecureRandom
 import java.{ util ⇒ ju }
-import javax.net.ssl.KeyManager
-import javax.net.ssl.SSLContext
-import javax.net.ssl.SSLContextSpi
-import javax.net.ssl.SSLEngine
-import javax.net.ssl.SSLException
-import javax.net.ssl.SSLServerSocketFactory
-import javax.net.ssl.SSLSessionContext
-import javax.net.ssl.SSLSocketFactory
-import javax.net.ssl.TrustManager
+import javax.net.ssl._
 
+import scala.collection.immutable
+import scala.collection.JavaConverters._
 import akka.http.scaladsl.HttpsConnectionContext
 import org.eclipse.jetty.alpn.ALPN
+import org.eclipse.jetty.alpn.ALPN.ServerProvider
+
+object ALPNRegistry {
+  def registerServerEngine(engine: SSLEngine)(decider: immutable.Set[String] ⇒ Option[String]): Unit =
+    ALPN.put(engine, new ServerProvider {
+      def select(protocols: ju.List[String]): String =
+        choose(protocols.asScala.toSet)
```

Comment:
But it's still an API and it's not important for performance so I'd rather keep it that way.